### PR TITLE
New resource and data source - rule set

### PR DESCRIPTION
**Pull request by erezweinstein5-zz**
## Rule set Resource and Data Source
- New resource - Rule set
- New data source - Rule set
- Examples added
- Rule set resource test
- Rule set data source test
- Relevant documentation
- Bumped SDK version (v1.3.0)

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-dome9/blob/master/CHANGELOG.md):
***
Output from acceptance tests:
```
=== RUN   TestAccDataSourceRuleSetBasic
--- PASS: TestAccDataSourceRuleSetBasic (2.77s)
=== RUN   TestAccResourceRuleSetBasic
--- PASS: TestAccResourceRuleSetBasic (3.79s)
```

### CODE REVIEW

**Review by uditamar — COMMENTED** [review]
@talsheldon @ayalganem please review it together please

**Review by talsheldon — CHANGES_REQUESTED** [review]
Seems alright
Just run `terraform_fmt_recursive`

**Comment by talsheldon** [comment]
@ayalganem  Please review this once you are done with the CR fixes

File: dome9/common/providerconst/const.go [hunk 1]
```diff
@@ -15,3 +15,5 @@ const (
 
 // The 16 regions Dome9 manages in AWS cloud account
 var AWSRegions = []string{"us_east_1", "us_west_1", "eu_west_1", "ap_southeast_1", "ap_northeast_1", "us_west_2", "sa_east_1", "ap_southeast_2", "eu_central_1", "ap_northeast_2", "ap_south_1", "us_east_2", "ca_central_1", "eu_west_2", "eu_west_3", "eu_north_1"}
+
```
Comment (talsheldon) [inline dome9/common/providerconst/const.go]: remove

**Review by talsheldon — APPROVED** [review]

**Review by ayalganem — APPROVED** [review]

**Pull request merged**